src/test/java/com/amazonaws/crypto/examples/FileStreamingExampleTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.crypto.examples;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+class FileStreamingExampleTest {
+
+    @Test
+    void testEncryptAndDecrypt() throws IOException {
+        final File tempFile = File.createTempFile("FileStreamingExampleTest-TempTestData", ".tmp");
```

Comment:
try-with-resources? Or at least a try-finally? Yes it's a temp file but it's still polite to delete it. :)

Reply:
I'm using deleteOnExit() a couple lines down, is that not enough?

Reply:
Oh I somehow managed to miss that. :)

I have some bad experience with `deleteOnExit` in long-running processes (leaking memory by building up a never-ending list of files to delete) so I wouldn't recommend using it in any library source code, but I think it's fine for tests.